collectors/persistentvolumeclaim.go

```diff
@@ -26,15 +26,34 @@ import (
 )
 
 var (
+	descPersistentVolumeClaimInfo = prometheus.NewDesc(
+		"kube_persistentvolumeclaim_info",
+		"Information about persistent volume claim.",
+		[]string{
+			"namespace",
+			"storageclass",
+			"persistentvolumeclaim",
+		}, nil,
+	)
 	descPersistentVolumeClaimStatusPhase = prometheus.NewDesc(
 		"kube_persistentvolumeclaim_status_phase",
-		"The phase the claim is currently in.",
+		"The phase the persistent volume claim is currently in.",
 		[]string{
 			"namespace",
+			"storageclass",
```

Comment:
My point was that only the info metric actually needs this label, then the storageclass can be joined onto this metric using the info metric. This metric only needs the labels that uniquely identify it, which is just namespace, persistentvolumeclaim and phase.